### PR TITLE
fix: Long links within tables

### PR DESCRIPTION
**Pull request by HeleenSG**
### Changes

- Fixes links that are too long within tables. Often pushing other important data outside of the screen. By adding a max width to the links we can prevent most of these cases. And by adding the option to have one sticky column. Currently implemented on "see details". 
- Updates table html where needed.
- Fixes double h1 on crisis room
- Fixes broken table overflow within crisis room
- Fixes icon alignment within tables
- Adds overflow on tables where they were missing
- Adds missing class "actions" on action columns where missing

### Long links notes

**Benefits:** 
Long links (and very long links) push important content out of the screen. It can be accessed by scrolling horizontally but we've received feedback on multiple occasions that users are unaware of the scrolling option and run into problems because of this. By setting a max width to links we can prevent most of these cases. Not all unfortunately as it will depend on the number of columns within the table and the other content that is present. 

**Downsides:** 
- Please note that max width doesn't work on tables or  table cells. It does work on the links within the tables but only if they are set to block elements. I've changed the links to inline-block elements as it seems to be the most scaleable option at the moment. This can result in some white space being added to the link as it will behave as a block element. 
- By setting a max width on links within tables they will not use up all available space if there is more space available then the max width that has been set. Which can result in links being rendered on multiple lines even when they could in theory be wider in that specific scenario. 

### Demo
Before: Action buttons are "missing" on the right side as they are rendered off screen. 
Screenshot 2024-10-23 at 17 25 13

After: Action buttons are within the screen (longer) depending on the rest of the content and available space. The links are being placed on multiple lines.
Screenshot 2024-10-23 at 17 25 21

### Fixes
Before:
Screenshot 2024-10-29 at 15 53 16

After: 
Screenshot 2024-10-29 at 16 17 07

### QA notes

_Please add some information for QA on how to test the newly created code._

---

### Code Checklist

- [ ] All the commits in this PR are properly PGP-signed and verified.
- [ ] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] I have checked the documentation and made changes where necessary.
- [ ] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/assets/css/components/table.scss
Comment (underdarknl) [inline rocky/assets/css/components/table.scss:86]: When introducing this, we should probably also add valign:top to the other cells to provide a more consistent expderience for the remainder of the table-row if multi-line wrapping occurs .

**Comment by underdarknl** [comment]
`
<style>
table td{
  border:1px solid #000;
  max-width:150px;  
  width:150px;  
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}
table td:hover,
table td.longer{
  
  position:relative;
  overflow: visible;
}
table td:hover div,
table td.longer div{
  position:absolute;
  top:0;
  z-index:2;
  background-color:#fff;
  border:1px solid #f00;
}
</style>
<table cellpadding="2" cellspacing="0">
  <tr><td>a</td><td>b</td><td>c</td></tr>
  <tr><td><div>eeeeeeeeextra eeeeeeeeextra eeeeeeeeextra lang</div></td><td>b</td><td>c</td></tr>
  <tr><td>a</td><td class="longer"><div>eeeeeeeeextra eeeeeeeeextra eeeeeeeeextra lang</div></td><td>c</td></tr>
</table>`

**Comment by underdarknl** [comment]
Code looks good to me, lets do QA.

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Looks good. Couldn't find obvious regression. Double checked a few things, like on the objects list longer domains are now shown on two rows instead of a single row, but that is expected with this fix. 

### What doesn't work:
n/a

### Bug or feature?:
@HeleenSG observed that Tree and Graph icons on the Findings page were not aligned during my QA, so waiting for the fix on that before it can be merged.

**Comment by HeleenSG** [comment]
- Fixed alignment of th elements on action icons: 
Screenshot 2024-11-04 at 11 54 37

- Added alignment top for cell content when displaying long links over multiple lines.

**Comment by stephanie0x00** [comment]
Issue resolved :+1:

**Review by underdarknl — APPROVED** [review]

**Review by underdarknl — APPROVED** [review]

**Pull request merged**
